File: EpiFib/mobile/EpiFib/EpiFib/EmergencyInstanceRequest.swift
//
//  EmergencyInstanceRequest.swift
//  EpiFib
//
//  Created by Gregory Hewett on 3/19/17.
//  Copyright © 2017 The Reach Lab, LLC. All rights reserved.
//

import UIKit
import EVReflection

final class EmergencyInstanceRequest: EVNetworkingObject
{
    var emergencyInstanceId: String?
    
    var emergencyType: EmergencyType?
    
    var ownerLocation: ExactLocation?
    
    var updatedTime: NSDate?
    
    var createdTime: NSDate?
    
    convenience init(_ emergencyType: EmergencyType, _ ownerLocation: ExactLocation)
    {
        self.init()
        self.emergencyType = emergencyType
        self.ownerLocation = ownerLocation
    }
}

File: EpiFib/www/src/Common/Models/EmergencyInstanceRequest.cs
// <copyright file="EmergencyInstanceRequest.cs" company="The Reach Lab, LLC">
//     The Reach Lab, LLC. All rights reserved.
// </copyright>
// <author>Gregory Hewett</author>

namespace Common.Models
{
    using System;

    public class EmergencyInstanceRequest : IEmergencyBase
    {
        public string EmergencyInstanceId { get; set; }

        public EmergencyType EmergencyType { get; set; }

        public ExactLocation OwnerLocation { get; set; }

        public EmergencyUserInfo OwnerInfo { get; set; }

        public DateTime UpdatedTime { get; set; }

        public DateTime CreatedTime { get; set; }

        public static explicit operator EmergencyInstanceRequest(EmergencyInstance v)
        {
            return new EmergencyInstanceRequest
            {
                EmergencyInstanceId = v.EmergencyInstanceId,
                EmergencyType = v.EmergencyType,
                OwnerLocation = v.OwnerLocation,
                UpdatedTime = v.UpdatedTime,
                CreatedTime = v.CreatedTime,
                OwnerInfo = v.OwnerInfo
            };
        }
    }
}
